### ISSUE TITLE
Feature request - rxjs rules from tslint

### ISSUE THREAD

**Issue by Lonli-Lokli** [typescript-eslint/typescript-eslint#533]
There is a set of rxjs related rules for tslint
https://github.com/cartant/rxjs-tslint-rules/blob/master/README.md
It might be very good to have same set for eslint.

**Comment by rassie** [comment]
There is an initial WIP effort to port those rules to ESLint by @cartant: https://github.com/cartant/eslint-plugin-rxjs.

**Comment by cartant** [comment]
Yeah. It'll happen. I have a few other things going on, ATM. I intend to write an article on how to build ESLint rules and, after that, if anyone wants to help convert the rules, it would be appreciated.

**Comment by bradzacher** [comment]
@cartant - you should consider using our `experimental-utils` package to help with writing an eslint plugin in typescript.

**Comment by cartant** [comment]
@bradzacher Thanks for the suggestion. That's really helpful.

A short while ago, I noticed the approach taken in [`ban-types.ts`](https://github.com/typescript-eslint/typescript-eslint/blob/5792aa1e45b7643e58f3187879c431ddd5b432f6/packages/eslint-plugin/src/rules/ban-types.ts) - which uses `@typescript-eslint/experimental-utils` - and was unsure of whether or not I should implement rules in a similar manner. The rules I've implemented thus far have used `esTreeNodeToTSNodeMap` to obtain the TS nodes.

**Comment by bradzacher** [comment]
It's called `experimental-utils` purely because we didn't want to make it a `0.x.y` package when everything else is `1.x.y`. Just treat minor version bumps as potentially breaking and you'll be fine with it.

That package should contain everything that we use to create rules within our plugin, so you can take advantage of the strict typing and utility functions too!

**Comment by cartant** [comment]
@bradzacher So using `esTreeNodeToTSNodeMap` rather than methods named [like this](https://github.com/typescript-eslint/typescript-eslint/blob/5792aa1e45b7643e58f3187879c431ddd5b432f6/packages/eslint-plugin/src/rules/ban-types.ts#L90):

```ts
'TSTypeReference Identifier'(node: TSESTree.Identifier) {
```

is preferred?

**Comment by bradzacher** [comment]
oh no sorry, you misunderstand me.
ESLint requires those listener functions. If you have no listener functions, eslint won't invoke your rule at all.

`esTreeNodeToTSNodeMap` is a map that our parser builds which you can interrogate with the nodes that eslint gives you. This map will return the equivalent nodes from the typescript parser.

**Comment by cartant** [comment]
@bradzacher Yeah. I understand that the listener functions are required.

I've translated a couple of the simpler rules and the one that uses the node map (and the type checker) is [here](https://github.com/cartant/eslint-plugin-rxjs/blob/8d19e39e0ae8094d207e19dc38da9c619a9b57d1/source/rules/no-connectable.ts#L24-L60).

The part that I'm not clear on is whether using listener names like `TSTypeReference Identifier` is something that's encouraged. Or whether that approach is internal and something that should not be depended upon, lest it change.

**Comment by bradzacher** [comment]
Oh sorry. Yeah that's what you need to do if you're looking to lint TS specific features.
The AST representation is designed and fixed and intended for use.

If we change the specification, we'll release a new major version of everything.
The types for the spec are exported: `import { TSESTree } from 'experimental-utils';`.

**Comment by cartant** [comment]
Awesome. Thanks.

**Comment by JamesHenry** [comment]
I'm going to close this as this is appropriately being developed as a community plugin using the utils we provide.

Thanks all 🙏
